src/app/utils/ScoapUtils.ts

```diff
@@ -0,0 +1,76 @@
+import constants from '../service/constants/constants';
+// import { GuildMember, Message } from 'discord.js';
+import fs from 'fs';
+import path from 'path';
+import { scoapEmbedState, botConvoState, voteRecordState } from '../service/scoap-squad/ScoapDatabase';
+
+const isInteger = (value: any): boolean => {
+	return /^\d+$/.test(value);
+};
+
+const ScoapUtils = {
+
+	validateTotalNumberOfRoles(message_content: string): boolean {
+		return (isInteger(message_content) && (parseInt(message_content) < 10) && (parseInt(message_content) >= 1));
+	},
+
+	validateTotalNumberOfPeoplePerRole(message_content: string): boolean {
+		return (isInteger(message_content) && (parseInt(message_content) < 1000) && (parseInt(message_content) >= 1));
+	},
+
+	validateSummary(summary: string): boolean {
+		const CREATE_SUMMARY_REGEX = /^[\w\s.!@#$%&,?']{1,4000}$/;
+		return (!(summary == null || !CREATE_SUMMARY_REGEX.test(summary)));
+	},
+
+	validateReward(message_content: string): boolean {
+		const [reward, symbol] = (message_content != null) ? message_content.split(' ') : [null, null];
+		const amount = parseInt(reward);
+		const ALLOWED_CURRENCIES = ['ETH', 'BANK'];
+		const MAXIMUM_REWARD = 100000000;
+		return (!(isNaN(amount) || amount <= 0 || amount > MAXIMUM_REWARD
+					|| !ALLOWED_CURRENCIES.includes(symbol)));
+	},
+
+	validateTitle(title: string): boolean {
+		const CREATE_TITLE_REGEX = /^[\w\s.!@#$%&,?']{1,250}$/;
+		return (!(title == null || !CREATE_TITLE_REGEX.test(title)));
+	},
+
+	getKeyByValue(object: any, value: string): any {
+		return Object.keys(object).find(key => object[key] === value);
+	},
+
+	purgeExpiredBotConvo(bot_convo_state) {
+		if (Object.keys(bot_convo_state).length === 0 && bot_convo_state.constructor === Object) {
+			// console.log('NO CONVO');
+			return;
+		}
+
+		for (const key of Object.keys(bot_convo_state)) {
+		    const dtnow = +new Date();
+			const dtold = bot_convo_state[key].getTimeout();
+			const deltat = dtnow - dtold;
+			if (deltat > constants.BOT_CONVERSATION_TIMEOUT_MS) {
+				bot_convo_state[key].getCurrentChannel().send('Conversation timed out. please try again.');
+				delete bot_convo_state[bot_convo_state[key].getUserId()];
+				this.logToFile(`object  deleted from botConvoState. reason: BotConvo timeout, deltaT: ${deltat} \n scoapEmbedState: ${scoapEmbedState} \n botConvoState: ${botConvoState}  \n voteRecordState: ${voteRecordState}`);
+			}
+		}
+		return;
+	},
+
+	logToFile(log_string: string): any {
+		const dt = new Date();
+		const formatted_str = '\n\n' + dt.toString() + '\n' + log_string;
+		const fPath = path.join(__dirname, '..', 'service', 'scoap-squad', 'logs', 'scoap-log');
+		fs.appendFile(fPath, formatted_str, function(err) {
+		    if(err) {
```

Comment:
The path for one of the function calls is throwing EOF error

```
2021-09-07T14:12:45.741472+00:00 app[worker.1]: Slinkypotato#1092 (159014522542096384) ran command scoap-squad
2021-09-07T14:12:45.741762+00:00 app[worker.1]: /scoap-squad assemble new
2021-09-07T14:12:45.749943+00:00 app[worker.1]: [Error: ENOENT: no such file or directory, open '/app/dist/app/service/scoap-squad/logs/scoap-log'] {
2021-09-07T14:12:45.749945+00:00 app[worker.1]: errno: -2,
2021-09-07T14:12:45.749946+00:00 app[worker.1]: code: 'ENOENT',
2021-09-07T14:12:45.749946+00:00 app[worker.1]: syscall: 'open',
2021-09-07T14:12:45.749947+00:00 app[worker.1]: path: '/app/dist/app/service/scoap-squad/logs/scoap-log'
2021-09-07T14:12:45.749947+00:00 app[worker.1]: }
2021-09-07T14:12:45.916263+00:00 app[worker.1]: debug: undefined 1631023965916 /interactions/884803384534773770/:token/callback 204: 172ms (234ms avg) | 1/1 left | Reset 1631023965916 (0ms left)
2021-09-07T14:12:50.411240+00:00 app[worker.1]: /scoap-squad end Slinkypotato#1092
```

We should assume that the dist directory  can be fully destroyed upon building across environments

Reply:
This is a temporary log file I just wanted to use for logging state during the testing phase to help debugging. I changed the log file path to log directly into the scoap-squad folder, hope that resolves the error.